### PR TITLE
Add merge(with:) operator

### PR DESCRIPTION
**Pull request by ra1028**
Refer https://github.com/ReactiveCocoa/ReactiveSwift/issues/597

#### Checklist
- [x] Updated CHANGELOG.md.

### CODE REVIEW

File: Sources/Signal.swift [hunk 1]
```diff
@@ -788,6 +788,17 @@ extension Signal {
...
+	/// Merge the given signal into a single `Signal` that will emit all
+	/// values from both of them, and complete when all of them have completed.
+	///
+	/// - parameters:
+	///   - other: A signal to merge `self`'s value with.
+	///
+	/// - returns: A signal that sends all values of `self` and given signal.
+	public func merge(with other: Signal<Value, Error>) -> Signal<Value, Error> {
+		return Signal.merge(self, other)
+	}
```
Comment (mdiep) [inline Sources/Signal.swift]: I think we still ought to have a test for this (and the `SignalProducer` variant below). Would you mind adding one?

File: Sources/Flatten.swift [hunk 2]
```diff
@@ -481,46 +481,32 @@ extension UnsafeAtomicState where State == ProducerState {
 }
 
 extension Signal {
-	/// Merges the given signals into a single `Signal` that will emit all
-	/// values from each of them, and complete when all of them have completed.
-	///
-	/// - parameters:
-	///   - signals: A sequence of signals to merge.
+	/// Merges the values of all the given signals, in the manner described
+	/// by `merge(with:)`.
```
Comment (NachoSoto) [inline Sources/Flatten.swift]: Normally I wouldn't mind this, but Xcode does not make it easy to find this documentation, so I'd keep it duplicated.

Comment (ra1028) [reply Sources/Flatten.swift]: hmm, I was according to `zip`, `combineLatest` operator, though should revert?

**Comment by ra1028** [comment]
I added tests

File: Sources/Flatten.swift (hunk 2 above)
Comment (NachoSoto) [reply Sources/Flatten.swift]: We should duplicate it in those too IMO (at least until SourceKit & CMD+Click actually works consistently in Xcode). For now can you keep these how they were?

File: Tests/ReactiveSwiftTests/FlattenSpec.swift [hunk 3]
```diff
@@ -820,8 +820,9 @@ class FlattenSpec: QuickSpec {
 			it("should emit values from all signals") {
 				let (signal1, observer1) = Signal<Int, NoError>.pipe()
 				let (signal2, observer2) = Signal<Int, NoError>.pipe()
+				let (signal3, observer3) = Signal<Int, NoError>.pipe()
 
-				let mergedSignals = Signal.merge([signal1, signal2])
+				let mergedSignals = Signal.merge([signal1, signal2, signal3])
```
Comment (NachoSoto) [inline Tests/ReactiveSwiftTests/FlattenSpec.swift]: Thanks for making this more complex now 👍

File: Sources/Flatten.swift (hunk 2 above)
Comment (ra1028) [reply Sources/Flatten.swift]: gotcha, will revert them:+1:
Write the doc for `zip` and` combineLatest`, will too much lines.
I think there are pros and cons, so not to write now.

**Comment by mdiep** [comment]
Awesome! Thanks for adding this. 🤘

**Pull request merged**